pkg/k8s/apis/cilium.io/v2alpha1/bgp_node_override_types.go

```diff
@@ -0,0 +1,100 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright Authors of Cilium
+
+package v2alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// +genclient
+// +genclient:nonNamespaced
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+// +kubebuilder:resource:categories={cilium,ciliumbgp},singular="ciliumbgpnodeconfigoverride",path="ciliumbgpnodeconfigoverrides",scope="Cluster",shortName={bgpnodeoverride}
+// +kubebuilder:printcolumn:JSONPath=".metadata.creationTimestamp",name="Age",type=date
+// +kubebuilder:storageversion
+
+// CiliumBGPNodeConfigOverride is used to overrides some of the BGP configurations which are node local.
+// Users can user this resource to override auto-generated BGP settings for the node.
+type CiliumBGPNodeConfigOverride struct {
+	// +deepequal-gen=false
+	metav1.TypeMeta `json:",inline"`
+	// +deepequal-gen=false
+	metav1.ObjectMeta `json:"metadata"`
+
+	// Spec is the specification of the desired behavior of the CiliumBGPNodeConfigOverride.
+	Spec CiliumBGPNodeConfigOverrideSpec `json:"spec"`
+}
+
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+// +k8s:openapi-gen=false
+// +deepequal-gen=false
+
+// CiliumBGPNodeConfigOverrideList is a list of CiliumBGPNodeConfigOverride objects.
+type CiliumBGPNodeConfigOverrideList struct {
+	metav1.TypeMeta `json:",inline"`
+	metav1.ListMeta `json:"metadata"`
+
+	// Items is a list of CiliumBGPNodeConfigOverride.
+	Items []CiliumBGPNodeConfigOverride `json:"items"`
+}
+
+type CiliumBGPNodeConfigOverrideSpec struct {
+	// NodeRef is the name of the node for which the BGP configuration is overridden.
+	//
+	// +kubebuilder:validation:Required
+	NodeRef string `json:"nodeRef"`
+
+	// BGPInstances is a list of BGP instances to override.
+	//
+	// +kubebuilder:validation:Required
+	// +kubebuilder:validation:MinItems=1
+	// +listType=map
+	// +listMapKey=localASN
+	BGPInstances []CiliumBGPNodeConfigInstanceOverride `json:"bgpInstances"`
+}
+
+// CiliumBGPNodeConfigInstanceOverride defines configuration options which can be overridden for a specific BGP instance.
+type CiliumBGPNodeConfigInstanceOverride struct {
+	// LocalASN is used to identify the BGP instance to override.
+	//
+	// +kubebuilder:validation:Required
+	LocalASN int64 `json:"localASN"`
+
+	// RouterID is BGP router id to use for this instance. It must be unique across all BGP instances.
+	// If not specified, the node's IP address will be used.
+	//
+	// +kubebuilder:validation:Optional
+	// +kubebuilder:validation:Format=ipv4
+	RouterID string `json:"routerID,omitempty"`
```

Comment:
Should this type be a pointer for consistency with other optional types? I see this elsewhere, so do a find/replace through all API types if you agree with this comment.